pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -391,6 +400,11 @@ func (b *volumeBinder) checkBindings(pod *v1.Pod, bindings []*bindingInfo, claim
 			return false, fmt.Errorf("failed to check pv binding: %v", err)
 		}
 
+		// Check PV's node affinity (the node might not have the proper label)
+		if err := volumeutil.CheckNodeAffinity(pv, node.Labels); err != nil {
+			return false, fmt.Errorf("pv %q node affinity doesn't match node %q: %v", pv.Name, node.Name, err)
+		}
```

Comment:
For static PV binding, PV has been checked in finding, this is used to double check before scheduling?

https://github.com/kubernetes/kubernetes/blob/98b3e4212797f87410ec8b12a836c289b2e68102/pkg/controller/volume/persistentvolume/index.go#L178-L186`

Reply:
Not because of the node labels, but because because of the new static binding [test added](https://github.com/kubernetes/kubernetes/pull/70026#discussion_r244380476).

Reply:
The check here is more of a safety guard in case the user ends up manually binding to a different PV on a different node than what the scheduler chose